src/scripts/main.js

```diff
@@ -1,3 +1,96 @@
 'use strict';
 
-// write code here
+const rowMin = 2;
+const rowMax = 10;
+const colMin = 2;
+const colMax = 10;
+const table = document.querySelector('.field');
+const buttonAddCol = document.querySelector('.append-column');
+const buttonRemoveCol = document.querySelector('.remove-column');
+const buttonAddRow = document.querySelector('.append-row');
+const buttonRemoveRow = document.querySelector('.remove-row');
+
+function addCol() {
+  const colsInitial = table.querySelector('tr')
+    .querySelectorAll('td').length;
+
+  if (colsInitial >= colMax) {
+    return;
+  }
+
+  if (colsInitial >= colMax - 1) {
+    buttonAddCol.disabled = true;
+  }
+
+  buttonRemoveCol.disabled = false;
+
+  for (const tr of table.querySelectorAll('tr')) {
+    const td = document.createElement('td');
+
+    tr.append(td);
+  }
+}
+
+function removeCol() {
+  const colsInitial = table.querySelector('tr')
+    .querySelectorAll('td').length;
+
+  if (colsInitial <= colMin) {
+    return;
+  }
+
+  if (colsInitial <= colMin + 1) {
+    buttonRemoveCol.disabled = true;
+  }
+
+  buttonAddCol.disabled = false;
+
+  for (const tr of table.querySelectorAll('tr')) {
+    const td = tr.lastElementChild;
+
+    td.remove();
+  }
+}
+
+function addRow() {
+  const rowsInitial = table.querySelectorAll('tr').length;
+
+  if (rowsInitial >= rowMax) {
+    return;
+  }
+
+  if (rowsInitial >= rowMax - 1) {
+    buttonAddRow.disabled = true;
+  }
+
+  buttonRemoveRow.disabled = false;
+
+  const tbody = table.querySelector('tbody');
```

Comment:
you can get tbody on head file one time and don't get it every time when you click add or remove row